### PR TITLE
Promote experimental out-of-process features to stable

### PR DESCRIPTION
**Pull request by sharwell**
This is the first of two ways to transition OOP features from experimental to stable.

* Features deemed stable by prior experiments are transitioned to stable, and will run out of process via the "normal" mechanism (as long as OOP itself is enabled) rather than guarded by `RemoteFeatureOptions.OutOfProcessEnabled`
* ~~The UI is not touched; the setting for enabling experimental OOP features remains in place and we can use it for future work in this space~~ Edit: The previous UI for experimental OOP features has been eliminated. The option is not enabled by default, with no exposed way to disable it.

See #27985 for the second option.

<details><summary>Ask Mode template not completed</summary>

### Customer scenario

What does the customer do to get into this situation, and why do we think this
is common enough to address for this release.  (Granted, sometimes this will be
obvious "Open project, VS crashes" but in general, I need to understand how
common a scenario is)

### Bugs this fixes

(either VSO or GitHub links)

### Workarounds, if any

Also, why we think they are insufficient for RC vs. RC2, RC3, or RTW

### Risk

This is generally a measure our how central the affected code is to adjacent
scenarios and thus how likely your fix is to destabilize a broader area of code

### Performance impact

(with a brief justification for that assessment (e.g. "Low perf impact because no extra allocations/no complexity changes" vs. "Low")

### Root cause analysis

How did we miss it?  What tests are we adding to guard against it in the future?

### How was the bug found?

(E.g. customer reported it vs. ad hoc testing)

### Test documentation updated?

If this is a new non-compiler feature or a significant improvement to an existing feature, update https://github.com/dotnet/roslyn/wiki/Manual-Testing once you know which release it is targeting.

</details>

### CODE REVIEW

File: src/Workspaces/Core/Portable/Remote/RemoteFeatureOptions.cs [hunk 1]
```diff
@@ -15,15 +14,30 @@ internal static class RemoteFeatureOptions
         /// Global switch that determines of OOP can be used for a language feature. Exposed through
         /// user visible switch to let people opt-in/out of this behavior.
         /// </summary>
-        public static readonly Option<bool> OutOfProcessAllowed = new Option<bool>(
+        public static readonly Option<bool> ExperimentalOutOfProcessAllowed = new Option<bool>(
             nameof(RemoteFeatureOptions), nameof(OutOfProcessAllowed), defaultValue: false,
             storageLocations: new LocalUserProfileStorageLocation(LocalRegistryPath + nameof(OutOfProcessAllowed)));
 
+        /// <summary>
+        /// This field contains the original name of the experimental OOP feature, which is used as the key for saving
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Remote/RemoteFeatureOptions.cs]: The pattern we have here is just to stop using nameof. TBH I consider using nameof to be an anti-pattern in synced options in general precisely for this reason...

**Comment by CyrusNajmabadi** [comment]
Just to check, but is htis promoting all the OOP feature to stable.  I ask because it somewhat needs to be all/nothing.  The reason for this is that if you miss a single feature, then you can get teh worst of all worlds, where both teh VS process and the OOP process need to fully load all compilations across all projects for global operations.  The work i did was to OOPify everything that would do this, making it so that VS should not ever need global information, and everything that needs global will be in the OOP process.

it would probably be good to ensure that nothing in the meantime has broken this :)

**Comment by sharwell** [comment]
@CyrusNajmabadi Yes, all current experimental features are promoted

**Comment by sharwell** [comment]
@jasonmalinowski This pull request could be updated to *hide* the experimental OOP check box since it won't have any behavior impact prior to the reintroduction (if ever) of experimental features.

**Comment by CyrusNajmabadi** [comment]
I would just remove the checkbox.  We originally had this because there was large concern about if this approach was viable, or if it would lead to significant unknown issues.  So we needed a way to be able to turn this all off simply.  I think such a need is no longer there.  The feature works and was effectively A/B tested.  Any future global-ops should just be OOP by default.

**Comment by jasonmalinowski** [comment]
Agreeing with @CyrusNajmabadi: just delete. I fully expect too that any OOP stuff won't be experimental: instead we'll be building it that way from day one. And if we want the checkbox back, `git revert` exists.

File: src/Workspaces/Core/Portable/Remote/RemoteHostClientExtensions.cs [hunk 2]
```diff
@@ -1,13 +1,12 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.CodeAnalysis.Experiments;
-using Microsoft.CodeAnalysis.Options;
 using Microsoft.CodeAnalysis.Execution;
 using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.Options;
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Remote/RemoteHostClientExtensions.cs]: Ugh, if it wasn't for the sort you could have had a perfect 100% delete PR. :smile:

**Comment by sharwell** [comment]
@jinujoseph for approval

**Comment by CyrusNajmabadi** [comment]
Oh man this makes me happy :)

**Comment by sharwell** [comment]
@jinujoseph This now has design agreement across four people with significant participation in the OOP features. I believe it is enough to move forward once you are ready.

**Comment by jinujoseph** [comment]
waiting for A/B report from Duy 
cc @Pilchie does this need QB approval too ?

**Comment by Pilchie** [comment]
No, not at this point as long as @dotnet/roslyn-infrastructure is prepared to get it in for 15.8P4.

**Comment by sharwell** [comment]
@jinujoseph for approval

**Comment by jinujoseph** [comment]
@sharwell can you spin a val build with this change pls.
This is pending approval on the latest results .

**Comment by sharwell** [comment]
@jinujoseph The val build and RPS came back green on this change.

**Comment by CyrusNajmabadi** [comment]
@jinujoseph @sharwell where are we in terms of getting this in?

**Comment by jinujoseph** [comment]
@sharwell can you retarget this change to 16.1 branch pls. 
Plan is to get this in next update.

**Comment by CyrusNajmabadi** [comment]
@sharwell @jinujoseph Can this go in now?

**Comment by CyrusNajmabadi** [comment]
Nice!

**Pull request merged**